lib/templates/components/content/CeDynamic.js

```diff
@@ -18,27 +18,50 @@ export default {
       default: 'text'
     }
   },
+  // content element components are registered in CamelCase like "CeMyComponent",
+  // but the type is coming as "my_component" so we have to transform the name and the tag
+  computed: {
+    componentName() {
+      const typeNameCamelCase = this.type.replace(/_([a-z])/g, g => {
+        return g[1].toUpperCase()
+      })
+
+      return `Ce${typeNameCamelCase[0].toUpperCase() +
+        typeNameCamelCase.slice(1)}`
+    },
+    componentTag() {
+      return `ce-${this.type.replace(/_/g, '-')}`
+    }
+  },
   render(createElement, ctx) {
-    // content element components are registered as "CeComponent" convention
-    // type is getting as "component"
-    // we have to concat it and check if component is registered
-    const componentName = `Ce${this.type[0].toUpperCase() + this.type.slice(1)}`
-    const createComponent = () =>
-      createElement(
-        this.$nuxt.$options.components[componentName]
-          ? `ce-${this.type}`
-          : 'ce-default',
-        {
-          props: {
-            ...{
-              id: this.data.uid,
-              type: this.data.type,
-              appearance: this.data.appearance
-            },
-            ...this.data.content
-          }
+    const createComponent = () => {
+      let elementTag
```

Comment:
What do you think about simplify this a bit ? I think we don't need computed values because these component tags won't changed anymore, this is rendering only once - when new page content is delivered by API.

What about something like this: 

```js
      // content element components are registered in CamelCase like "CeMyComponent",
      // but the type is coming as "my_component" so we have to transform the name and the tag
      const typeNameCamelCase = this.type.replace(/_([a-z])/g, g =>
        g[1].toUpperCase()
      )

      let elementTag = `Ce${typeNameCamelCase[0].toUpperCase() +
        typeNameCamelCase.slice(1)}`

      if (!this.$nuxt.$options.components[elementTag]) {
        // fallback, try the old name and tag style (TODO: can be removed in future "breaking changes" version)
        const componentName = `Ce${this.type[0].toUpperCase() +
          this.type.slice(1)}`

        if (this.$nuxt.$options.components[componentName]) {
          elementTag = `ce-${this.type}`
        } else {
          elementTag = 'ce-default'
        }
      }
```

This change is very simple and thanks to that we can get rid of ugly naming, thank you 👍